test/integration/host_js_test.rb

```diff
@@ -22,6 +22,34 @@ class HostJSTest < IntegrationTestWithJavascript
     SETTINGS[:organizations_enabled] = true
   end
 
+  describe 'multiple hosts selection' do
+    setup do
+      FactoryBot.create(:host)
+      FactoryBot.create(:host)
+    end
+
+    test 'hosts counter should refer to per_page value first (max prespective)' do
+      Setting[:entries_per_page] = 2
+      visit hosts_path(per_page: 3)
+      check 'check_all'
+      assert page.has_text?(:all, "All 3 hosts on this page are selected")
```

Comment:
if you are creating only 2 hosts how are you sure there are more then 3?  I'm assuming this works now because of the hosts fixture, but if that ever changes it will break this test in an unexpected way. Please create enough hosts for the test you need.

Reply:
It's not related to hosts fixtures, one host is created before each test (as you can see at line 17).